### PR TITLE
fix: produce error instances from xhr error & timeout event callbacks

### PR DESCRIPTION
**Pull request by bjoerge**
Currently the `xhr.onerror` and `xhr.ontimeout` events are forwarded to the error callback as-is (i.e. [ProgressEvent instances](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/timeout_event#event_type)). This breaks the widespread expectation that a thrown error is always an instance of `Error`.

This PR adds event callbacks that converts the progress events into error instances before passing on to the `onError`-callback.

### CODE REVIEW

File: src/request/browser-request.ts [hunk 1]
```diff
@@ -40,8 +40,24 @@ export default (context: any, callback: any) => {
...
   // Apply event handlers
-  xhr.onerror = onError
-  xhr.ontimeout = onError
+  xhr.onerror = (event: ProgressEvent) => {
+    onError(
+      new Error(
+        `Request error while attempting to reach ${options.url}${
+          event.lengthComputable ? `(${event.loaded} of ${event.total} bytes transferred)` : ''
+        }`
+      )
```
Comment (stipsan) [inline src/request/browser-request.ts:49]: It'd be nice to pass [cause](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause) so the original event can still be accessed:

```suggestion
      new Error(
        `Request error while attempting to reach ${options.url}${
          event.lengthComputable ? `(${event.loaded} of ${event.total} bytes transferred)` : ''
        }`,
        {cause: event}
      )
```

File: src/request/browser-request.ts [hunk 2]
```diff
@@ -40,8 +40,24 @@ export default (context: any, callback: any) => {
...
+    )
+  }
+  xhr.ontimeout = (event: ProgressEvent) => {
+    onError(
+      new Error(
+        `Request timeout while attempting to reach ${options.url}${
+          event.lengthComputable ? `(${event.loaded} of ${event.total} bytes transferred)` : ''
+        }`
+      )
```
Comment (stipsan) [inline src/request/browser-request.ts:58]:
```suggestion
      new Error(
        `Request timeout while attempting to reach ${options.url}${
          event.lengthComputable ? `(${event.loaded} of ${event.total} bytes transferred)` : ''
        }`,
        {cause: event}
      )
```

**Review by stipsan — APPROVED** [review]
LGTM, it would be nice to use the `cause` property for `throw new Error` though ☺️

**Review by rexxars — APPROVED** [review]
Nice, thanks!

**Pull request merged**